lib/phoenix_live_view/test/live_view_test.ex

```diff
@@ -1028,6 +1034,125 @@ defmodule Phoenix.LiveViewTest do
     end
   end
 
+  @doc """
+  Open the default browser to display current HTML of `view_or_element`.
+
+  ## Examples
+
+      view
+      |> element("#term a:first-child()", "Increment")
+      |> open_browser()
+
+      assert view
+      |> form("#term", user: %{name: "hello"})
+      |> open_browser()
+      |> render_submit() =~ "Name updated"
+
+  """
+  def open_browser(view_or_element, open_fun \\ &open_with_system_cmd/1)
+
+  def open_browser(%View{} = view, open_fun) when is_function(open_fun, 1) do
+    view_html = render_tree(view, {proxy_topic(view), "render"})
+
+    view
+    |> maybe_wrap_html(view_html)
+    |> write_tmp_html_file()
+    |> open_fun.()
+
+    view
+  end
+
+  def open_browser(%Element{} = element, open_fun) when is_function(open_fun, 1) do
+    element_html = render_tree(element, element)
+
+    element
+    |> maybe_wrap_html(element_html)
+    |> write_tmp_html_file()
+    |> open_fun.()
+
+    element
+  end
+
+  def open_browser(html, %View{} = view) do
+    open_browser(html, view, &open_with_system_cmd/1)
+  end
+
+  @doc """
+  Open the default browser to display current `html` binary.
+
+  ## Examples
+
+      view
+      |> element("buttons", "Increment")
+      |> render_click()
+      |> open_browser()
+
+      assert view
+      |> form("#term", user: %{name: "hello"})
+      |> render_submit()
+      |> open_browser(view) =~ "Name updated"
+
+  """
+  def open_browser(html, %View{} = view, open_fun)
```

Comment:
I can see the purpose but I find the signature that receives both `html` and `%View{}` confusing for now. What do you think about removing it and keeping only the element/view ones as starting point?

Reply:
I was writing about it at the same time you sent this comment :D

It's useful because to have `open_browser(binary)` because you may want to see the result of a click, submit or similar. But I'm not happy with this API either, because it conflicts with the view/element and the doc looks weird.

I'd like to keep it if possible, maybe with another name, I'm not sure.

Reply:
Looks like that's the last piece to fix. I'm good with removing it for now, I just feel the feature may not be 100% complete if we remove it, but we could see how it goes without it.

Reply:
Adding features is always easy, removing is always hard, so I prefer to add it only if we need it. :)